### PR TITLE
Add new FelixConfiguration knobs for disabling encap drop rules

### PR DESCRIPTION
**Pull request by doublek**
## Description

Relates to 
- projectcalico/felix#2484
- docs: projectcalico/calico#3961

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Add FelixConfiguration parameters to explicitly allow encapsulated packets from workloads.
```

### CODE REVIEW

File: lib/apis/v3/felixconfig.go [hunk 1]
```diff
@@ -180,6 +180,13 @@ type FelixConfigurationSpec struct {
 	VXLANPort *int `json:"vxlanPort,omitempty"`
 	VXLANVNI  *int `json:"vxlanVNI,omitempty"`
 
+	// AllowVXLANPacketsFromWorkloads controls whether Felix will add a rule to drop VXLAN encapsulated traffic
+	// from workloads [Default: false]
+	AllowVXLANPacketsFromWorkloads *bool `json:"allowVXLANPacketsFromWorkloads,omitempty"`
+	// AllowIPIPPacketsFromWorkloads controls whether Felix will add a rule to drop IPIP encapsulated traffic
+	// from workloads [Default: false]
+	AllowIPIPPacketsFromWorkloads *bool `json:"allowIPIPPacketsFromWorkloads,omitempty"`
```
Comment (fasaxc) [inline lib/apis/v3/felixconfig.go:188]: Does it need to be a `*bool` if `false` is the default?  (Or maybe that's just our convention?)

**Review by fasaxc — APPROVED** [review]
One question but LGTM

File: lib/apis/v3/felixconfig.go (hunk 1 above)
Comment (fasaxc) [reply lib/apis/v3/felixconfig.go:188]: Oh, guess it's for inheritance, never mind.

File: lib/apis/v3/felixconfig.go [hunk 2]
```diff
@@ -180,6 +180,13 @@ type FelixConfigurationSpec struct {
 	VXLANPort *int `json:"vxlanPort,omitempty"`
 	VXLANVNI  *int `json:"vxlanVNI,omitempty"`
 
+	// AllowVXLANPacketsFromWorkloads controls whether Felix will add a rule to drop VXLAN encapsulated traffic
+	// from workloads [Default: false]
+	AllowVXLANPacketsFromWorkloads *bool `json:"allowVXLANPacketsFromWorkloads,omitempty"`
```
Comment (caseydavenport) [inline lib/apis/v3/felixconfig.go:185]: Now that we're using Kubernetes CRD generation with schemas, I think we need to explicit about defining optional fields and defaults.

So, this field should have comments like this on it in order to generate the correct CRDs. 

```
// +optional
// kubebuilder:default:=false
```

Source: https://book.kubebuilder.io/reference/markers/crd-validation.html

**Pull request merged**